feat: create slate-connector-gsuite

nbey opened the pull request:
- feat: implement gsuite connector
    - [x] implement gsuite users sync workflow
    - [x] implement gsuite calendar event creator workflow
    - [x] replace slate-cbl holosource with slate holosource
    - [x] detail service account requirements (gsuite service account w/ domain wide delegation)
        - clean up overrides
        - [x] `Slate\Logger` (remove after `\Emergence\Logger::interpolate()` is patched) (https://github.com/JarvusInnovations/emergence-skeleton/pull/237)
        - [x] `synchronizeComplete.tpl` (see above)
        - [x] `courseSection.tpl` (remove after `linksEntry` template is patched) (https://github.com/SlateFoundation/slate/pull/223)
    - decide which override to keep temporarily
        - [x] `CourseSection.js` (includes override to include required `Site.widget.GoogleCalendarEventCreator`)
       ~~- [ ] `courseSection.tpl` (_would_ include override to load `GoogleCalendarEventCreator`)~~

discussion:
- CreateEvent command
- PatchUser/CreateUser command?

nbey committed:
feat: create slate-connector-gsuite

nbey committed:
feat: add patch/create user API methods

themightychris commented on php-config/Slate/Connectors/GSuite/API.config.d/google-credentials.php:
```diff
@@ -0,0 +1,20 @@
+API::$clientId = '116995110434775517191';
+API::$clientEmail = 'calendar@calendar-project-272323.iam.gserviceaccount.com';
+API::$privateKey = "-----BEGIN PRIVATE KEY-----\nMIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQCjCuugaq6Z372n\n3DKogCVmUZnVz/35mQ8x6W1LBfkN77R5OlwWjodvkjGeRVzYWvy1G+wcl6Y0GCcf\nzAPdH4hrrgzpMhmUPlGTq8cKk9htTyaV+JDyb7sXLIRvBPt4h8rpuzqSLn5+wGAq\no/RRoBTWRAnfU833sn1/SVzz5uNz1uGQE0mog+LlL0rJdWhI0N9edxvO4pR2dNGY\n/T6TBi3N5zkuUhH9wCmrnz15q2nBg6wmsvyN5r3NBXRVBbMGp0UJQeK20nsN2rLV\nRTo7Zhry/h2aCaKvy9mP ...
+API::$domain = 'nafis.me';
```
let's blank all these out, maybe leave commented-out lines with example placeholders that help people see what the values should look like

nbey replied:
this wasn't supposed to be committed :/

nbey replied:
Ok, these credentials have been removed from their respective service accounts.

themightychris commented on php-config/Slate/UI/SectionProfile.config.d/gsuite-tools.php:
```diff
@@ -0,0 +1,19 @@
+    $links = [];
+
+    if (!empty($GLOBALS['Session']) && $GLOBALS['Session']->hasAccountLevel('Staff')) {
+        $sectionStudentEmails = array_filter(array_map(function($Student) {
```
everything looks really well-done, but I wonder if now that we're creating the calendar event from the backend, we could avoid loading the list of students on every page load and instead either:

- have the JS make an XHR call to load the section roster
- have the create event call pass a section id or student list string to the create event handler

themightychris reviewed (changes requested):
looking amazing so far, really well done overall! Couple tweaks I called out, and I responded to or merged the other PRs, so let's move forward getting the overrides resolved

nbey committed:
fix: remove keys from gsuite config

nbey committed:
chore: remove overrides

nbey committed:
chore: clean up code

nbey committed:
fix: add .studiorc config

nbey committed:
chore: switch slate-cbl source to slate

nbey committed:
chore: remove override

nbey committed:
feat: dynamic event attendees

- allow custom hidden form fields
- add `students` filter for adding event attendees

nbey committed:
chore: add readme

- detail service account authorization requirements
